src/engine/Match_search_mode.ml

```diff
@@ -523,17 +523,18 @@ let children_explanations_of_xpat (env : env) (xpat : Xpattern.t) : ME.t list =
 
 let rec filter_ranges (env : env) (xs : RM.ranges) (cond : R.metavar_cond) :
     RM.ranges =
+  let map_bool r b = if b then [ r ] else [] in
   xs
-  |> List.filter (fun r ->
+  |> List.concat_map (fun r ->
          let bindings = r.RM.mvars in
          match cond with
          | R.CondEval e ->
              let env = Eval_generic.bindings_to_env env.xconf.config bindings in
-             Eval_generic.eval_bool env e
+             Eval_generic.eval_bool env e |> map_bool r
          | R.CondNestedFormula (mvar, opt_lang, formula) ->
              (* TODO: could return expl for nested matching! *)
              Metavariable_pattern.satisfies_metavar_pattern_condition
-               nested_formula_has_matches env r mvar opt_lang formula
+               get_nested_formula_matches env r mvar opt_lang formula
```

Comment:
Wait, this is returning the matches of `metavariable-pattern`!

Reply:
This can lead to a significant change in semantics besides just allowing you to refer metavariables bound inside `metavariable-pattern`.

Reply:
I imagined this quite differently, that you would add to `r` the metavariables bound by `metavariable-pattern`.

Reply:
Oh OK, never mind me, sorry, that is what you did indeed! But the logic is in `Metavariable_pattern`.